src/react-components/src/components/02-molecules/menus/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -19,32 +20,36 @@ export default class Breadcrumb extends React.Component<BreadcrumbProps, {}> {
   }
 
   render(): JSX.Element {
-
     const { breadcrumbs } = this.props;
 
     let breadcrumbs__base_class = "breadcrumbs";
 
     let breadcrumbItems = [];
 
+    if (!breadcrumbs || breadcrumbs.length === 0) {
+      throw new Error("Breadcrumbs must contain a set of links. Breadcrumbs currently empty");
+    }
     breadcrumbs.forEach((item: breadcrumb|JSX.Element, index: number) => {
       let last = index === breadcrumbs.length - 1;
-      if (last) {
-        // TODO: SFR-716 Incorporate Icon
-      }
       let linkComponent: string | JSX.Element;
+
       if (isTextBreadcrumb(item)) {
         linkComponent = item.url ? <a href={item.url} className={bem("link", [], breadcrumbs__base_class)}>{item.text}</a> : item.text;
       } else {
         let props = {
-          className: bem("link", [], breadcrumbs__base_class),
+          className: bem("link", [], breadcrumbs__base_class)
```

Comment:
This is the second time `bem("link", [], breadcrumbs__base_class)` appears. Either make that a variable or maybe make a curry function (or HOC) function since many of the variables are constant. Kinda like
```
const withBreadcrumbsBaseClass = (type: string) => (
  return bem(type, [], breadcrumbs__base_class);
);

className={withBreadcrumbsBaseClass("link")}
```
but maybe a shorter name?

Reply:
bem already is a helper class that takes a bit of thinking to wrap the head around.  I'm hesitant to give it other helpers because it'll make it harder to associate the compiled classname with where it's used and where to make CSS changes (Unless we come up with a better source-mapping solution, which ... 🤔 ) 

I wonder if the frustration with this is more with how long `breadcrumbs__base_class` is as a variable name, making everything look ugly?  I don't mind assigning it to a variable, but on the other hand it doesn't seem that bad to me that this is being calculated twice - it's string concatenation without renders in between, which makes it very cheap, right?

Reply:
I would agree with Crystal here—I think the code duplication is fine as it maps to the way our Twig templates are working.

Reply:
Yeah, that makes sense, but this would be more of a helper just for this component since the modifiers and block name params don't change. It seems only the base class changes. It's definitely not necessary but I find HOCs to be easier to work with, personally. Those are cheap so it's fine.

Reply:
I guess the question then is if the helper should be abstracted, as this isn't the only component with this use case. If it's all right I think this is a really good conversation that would be aided with more data. I've added a ticket to the backlog [here](https://github.com/NYPL/nypl-design-system/projects/2#card-29725568) to look and some more use cases and to figure out how we should move forward, but let's not block this ticket on it.